cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -428,4 +422,45 @@ private InstanceId createInstanceId(CConfiguration cConf) {
                                 Constants.INSTANCE_NAME, instanceName);
     return new InstanceId(instanceName);
   }
+
+  private void validateNoMappingUpdates(NamespaceConfig existingConfig,
+                                        NamespaceConfig updatedConfig) throws BadRequestException {
+    if (updatedConfig == null) {
+      // nothing to validate
+      return;
+    }
+
+    String updatedRootDirectory = updatedConfig.getRootDirectory();
```

Comment:
Only, the error message will not be as accurate with a ``boolean equalsMapping``. I can create a ``Set<String> getDifference`` method, which might help.